### DISCUSSION TITLE
Deploy insiders with Cloudflare pages

### DISCUSSION THREAD

**Discussion by snjallingur** [squidfunk/mkdocs-material#6603] [General]
Hi,
we have purchased a license of insiders. A great addition for sure!
Our process is to deploy mkdocs through Cloudflare. So in our repro we have a requirement.txt file with all the pip modules to deploy. How should my requirements file look like?
I guess 

`mkdocs-material==9.5.3+insiders.4.49.0`

will not work?

**Comment by alexvoss** [comment]
Hi, thanks for sponsoring! The [publishing guide](https://squidfunk.github.io/mkdocs-material/publishing-your-site/#with-github-actions) recommends to install via 

```bash 
pip install git+https://${GH_TOKEN}@github.com/squidfunk/mkdocs-material-insiders.git
```

Check out the full example given on how to set this up and how to get the `GH_TOKEN` from your secrets.

**Comment by snjallingur** [comment]
Thanks for the reply. I actually found a 3rd party guide for [deploying through Cloudflare](https://www.starfallprojects.co.uk/projects/deploy-host-docs/deploy-mkdocs-material-cloudflare/#deployment). Then I was experiementing with `pip freeze` which revealed something like 

> mkdocs-material @ git+https://github.com/squidfunk/mkdocs-material-insiders.git@somegeneratedstring

was hoping this would be enough. However, `github.com/squidfunk/mkdocs-material-insiders.git` is not available at the moment?

**Reply by kamilkrzyskow** [reply]
The insiders repository is available to sponsors: https://squidfunk.github.io/mkdocs-material/insiders/
You either need to use an access token in that `git+...` part as Alex said above, or use an account with granted permissions in your Git client to access the repository. 
For automated deployments the token fetched from the secrets would be easier for sure: https://www.starfallprojects.co.uk/projects/deploy-host-docs/deploy-mkdocs-material-cloudflare/#deployment
So you need to use a separate command to install `mkdocs-material` from Insiders, as your `requirements.txt` probably won't process the `${GH_TOKEN}` part. However, you could also stick to the `requirements.txt` but pre-process it during deployment with replacing some placeholder with the actual token from the secrets.

Also the `@somegeneratedstring` part refers to a commit hash to lock the version.

**Reply by alexvoss** [reply]
It is possible to [include `GH_TOKEN` in `requirements.txt`](https://pip.pypa.io/en/stable/reference/requirements-file-format/#using-environment-variables) by using `${GH_TOKEN}`. Just came across this while searching for what would have been a more convoluted solution.

**Reply by kamilkrzyskow** [reply]
Oh wow, so text files do process variables now, or pip does it in this instance 😵‍💫😲 I'm impressed with the advancements in technology.

**Reply by alexvoss** [reply]
I think it will be the next big thing after all that AI hype ;o)

**Comment by snjallingur** [comment]
I simply added 

`git+https://myGHtoke@github.com/squidfunk/mkdocs-material-insiders.git@9.5.3-insiders-4.49.2`

to the requirements.txt file and now it is building fine. I guess the reason it didn't work earlier was because of the repository being unavailable (got a 404 error when running the pip install command).

Thanks for contributing.
I will mark this issue as resolved.

**Reply by alexvoss** [reply]
I would recommend keeping the secret out of the `requirements.txt` as a matter of good practice. Replace it with `${GH_TOKEN}` and make sure to have the token in the environment variable. If you are running CI then this should come out of the appropriate secrets collection. If you are running locally you can add it to a shell startup script or modify a virtual environment to set/unset it as you enter/exit the environment. (I did that but was using too many short-lived venvs to make this viable).
